apps/machine/microblaze_generic/platform_info.c

```diff
@@ -193,10 +193,10 @@ int platform_poll(void *priv)
 	return 0;
 }
 
-void platform_release_rpmsg_vdev(struct rpmsg_device *rpdev)
+void platform_release_rpmsg_vdev(struct rpmsg_virtio_device *rpvdev)
 {
-	if (rpdev->
-	(void)rpdev;
+	rproc_virtio_remove_vdev(rpvdev->vdev);
+	rpmsg_deinit_vdev(rpvdev);
```

Comment:
i think the order should be reversed  and using remoteproc_remove_virtio seems more clean.

```
rpmsg_deinit_vdev(rpvdev);
remoteproc_remove_virtio(rproc, rpvdev->vdev);

```

Reply:
so we can rproc from the platform, but otherwise how do you suggest to get that in? also is not rproc cleaned up later on?

Reply:
ok then we can add the demos' platform var which is of type rproc IIRC